TeslaMateAgile/Services/EnerginetService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.Extensions.Options;
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using TeslaMateAgile.Data;
+using TeslaMateAgile.Data.Options;
+using TeslaMateAgile.Services.Interfaces;
+
+namespace TeslaMateAgile.Services;
+
+public class EnerginetService : IPriceDataService
+{
+    private readonly HttpClient _client;
+    private readonly EnerginetOptions _options;
+    private readonly FixedPriceOptions _fixedPricesOptions;
+    private readonly FixedPriceService _fixedPriceService;
+
+    public EnerginetService(HttpClient client, IOptions<EnerginetOptions> options)
+    {
+        _client = client;
+        _options = options.Value;
+
+        _fixedPricesOptions = _options.FixedPrices;
+        _fixedPriceService = new FixedPriceService(Options.Create(_fixedPricesOptions));
```

Comment:
Currently the `FixedPrices` option must be set or it will crash during initialisation of this service. It should handle not being set and gracefully fall back. A good way to do this would be to check if `_fixedPricesOptions` is null and skip creating the service and any bit of code that uses it